### PR TITLE
make sure that we don't set a nil value for session id

### PR DESCRIPTION
**Pull request by derrickstaten**
@danwalkowski @aaustin

### CODE REVIEW

**Comment by derrickstaten** [comment]
for reference, https://github.com/BranchMetrics/iOS-Deferred-Deep-Linking-SDK/pull/218

**Comment by aaustin** [comment]
:+1:

File: Branch-SDK/Branch-SDK/Requests/BranchSetIdentityRequest.m [hunk 1]
```diff
@@ -57,7 +57,9 @@ - (void)processResponse:(BNCServerResponse *)response error:(NSError *)error {
     preferenceHelper.identityID = response.data[BRANCH_RESPONSE_KEY_BRANCH_IDENTITY];
     preferenceHelper.userUrl = response.data[BRANCH_RESPONSE_KEY_USER_URL];
     preferenceHelper.userIdentity = self.userId;
-    preferenceHelper.sessionID = response.data[BRANCH_RESPONSE_KEY_SESSION_ID];
+    if (response.data[BRANCH_RESPONSE_KEY_SESSION_ID]) {
```
Comment (shortstuffsushi) [inline Branch-SDK/Branch-SDK/Requests/BranchSetIdentityRequest.m]: Why would this be possible? This seems to indicate a problem elsewhere. Kinda similar to that type checking thing, it doesn't seem like the best idea to just wrap everything in if statements. I know it's just the one now, but you could really do the same for any of these.

Comment (derrickstaten) [inline Branch-SDK/Branch-SDK/Requests/BranchSetIdentityRequest.m]: We only sent down a session_id if it had changed. So this was a unique situation. (Last night to remedy this in realtime, we started sending down session_id 100% of the time.)

Comment (shortstuffsushi) [inline Branch-SDK/Branch-SDK/Requests/BranchSetIdentityRequest.m]: :+1: to that. That seems like the more consistent solution.

**Comment by danwalkowski** [comment]
I also :+1: this pr, and the 'always return a sessionID' behavior

**Pull request merged**